rest-service/manager_rest/rest/resources_v3_1/searches.py

```diff
@@ -187,16 +193,36 @@ class NodesSearches(ResourceSearches):
     @rest_decorators.sortable(models.Node)
     @rest_decorators.all_tenants
     @rest_decorators.search('id')
-    @rest_decorators.filter_id
     def post(self, _include=None, pagination=None, sort=None,
-             all_tenants=None, search=None, filter_id=None, **kwargs):
+             all_tenants=None, search=None, **kwargs):
         """List Nodes using filter rules or DSL constraints"""
-        deployment_id = retrieve_deployment_id()
+        deployment_id, _ = retrieve_deployment_id_and_constraints()
         return super().post(models.Node, None, _include,
                             {'deployment_id': deployment_id}, pagination,
                             sort, all_tenants, search, None, **kwargs)
 
 
+class NodeTypesSearches(ResourceSearches):
+    @swagger.operation(**_swagger_searches_docs(models.Node, 'nodes'))
+    @authorize('node_list', allow_all_tenants=True)
+    @rest_decorators.marshal_with(models.Node)
+    @rest_decorators.paginate
+    @rest_decorators.sortable(models.Node)
+    @rest_decorators.all_tenants
+    @rest_decorators.search('type')
+    def post(self, _include=None, pagination=None, sort=None,
```

Comment:
Why is the search a post rather than a get? I'm guessing something to do with the size of the parameters?

Reply:
yeah, all the searches are :/

Reply:
yeah, that's the case (sometimes)